xcube_stac/store.py

```diff
@@ -0,0 +1,337 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Any, Tuple, Iterator, Dict, Container, Union
+
+import logging
+import xarray as xr
+
+from xcube.util.jsonschema import (
+    JsonObjectSchema,
+    JsonStringSchema
+)
+from xcube.core.store import (
+    DATASET_TYPE,
+    DataDescriptor,
+    DataStore,
+    DataStoreError,
+    DataTypeLike
+)
+from .constants import DATASET_OPENER_ID
+from .opener import StacDataOpener
+from .stac import Stac
+
+_LOG = logging.getLogger("xcube")
+
+
+class StacDataStore(StacDataOpener, DataStore):
+    """ STAC implementation of the data store.
+
+    Attributes:
+        **stac_kwargs: Parameters required by the STAC data store.
+            * url: URL to STAC catalog (required)
+            * collection_prefix: Path of collection used as
+                entry point (optional)
+            * data_id_delimiter: Delimiter used to separate
+                collections, items and assets from each other (optional)
+    """
+
+    def __init__(self, **stac_kwargs):
```

Comment:
Please list all keywords here explicitly. In general, we should use `**` only for good reasons:

- The class/function is not public API
- The `kwargs` are dynamic and cannot be statically enumerated
- The `kwargs` are many and are documented elsewhere, so we can point users there

Reply:
I listed all keywords explicitly.